### PR TITLE
HPCC-16503 Generate constant inline child datasets efficiently

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please consider for 6.2.0.  It allows constant datasets with child datasets to be generated as constant definitions.

**Comment by ghalliday** [comment]
Please delay merging - I need to double check something

**Comment by ghalliday** [comment]
Just found a nasty lurking bug - could possibly be causing problems with existing code.

**Comment by ghalliday** [comment]
@richardkchapman should be good to review now.  Test case added to ensure it is generated correctly.

File: ecl/hql/hqlutil.cpp [hunk 1]
```diff
@@ -8545,18 +8546,41 @@ bool ConstantRowCreator::processFieldValue(IHqlExpression * optLhs, ITypeInfo *
...
                     {
                         rtlWriteSize32t(out.reserve(sizeof(size32_t)), 0);
                         memset(out.reserve(sizeof(byte * *)), 0, sizeof(byte * *));
+                        return true;
                     }
-                    else
+                }
+                else
+                {
+                    switch (rhs->getOperator())
```
Comment (richardkchapman) [inline ecl/hql/hqlutil.cpp]: Should this say rhsOp?

**Comment by richardkchapman** [comment]
@ghalliday One minor comment

**Comment by ghalliday** [comment]
@richardkchapman fixed

**Pull request merged**